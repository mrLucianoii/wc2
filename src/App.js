import React, { Component } from 'react';
import { connect } from 'react-redux';
import { isLoadingAction as actions } from './store/actions';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  componentDidMount(){
    const { loaderAction } = this.props;
    loaderAction();
  }
  render() {
    const { loading } = this.props;
    return (
      <div className="App">
      {!loading && <h1>Loading...</h1>}
       {loading && <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          WeCliques say's, "Hello World"
        </header>}
      </div>
    );
  }
}

export default connect(
  (state) => ({
    loading: state.featureReducer.isLoading,
  }),
  (dispatch) => ({
    loaderAction: () => dispatch(actions()),
  })
)(App);
