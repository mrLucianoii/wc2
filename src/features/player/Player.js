import React, { Component, PropTypes } from "react";
import { connect } from "react-redux";
import actions from "./store/actions";
import { Button } from "react-bootstrap";
import YouTube from "react-youtube";

export class Player extends Component {
  state = {
    player: null,
    duration: 0,
    segmentList: []
  };
  onReady = this.onReady.bind(this);
  onPlayVideo = this.onPlayVideo.bind(this);
  onPauseVideo = this.onPauseVideo.bind(this);

  onReady(event) {
    this.setState({
      player: event.target
    });
  }
  onPlayVideo() {
    const { startPlaying } = this.props.actions;
    const { selectedMovie, segmentStart } = this.props.movies;

    this.state.player.playVideo();
    startPlaying(selectedMovie);
  }
  seekSegment() {
    const { segmentStart } = this.props.movies;
    this.state.player.seekTo(segmentStart);
  }
  onPauseVideo() {
    this.state.player.pauseVideo();
  }

  render() {
    const { selectedMovie, segmentStart, segmentEnd } = this.props.movies;
    const { markInAction, markOutAction, checkMarks } = this.props.actions;
    const opts = {
      playerVars: {
        autoplay: 1,
        loop: 1,
        start: Number(segmentStart, 10),
        end: Number(segmentEnd, 10),
        modestbranding: 1,
        controls: 1,
        rel: 0
      }
    };
    return (
      <div style={{ width: 660, height: "auto" }}>
        <YouTube
          videoId={selectedMovie}
          onReady={this.onReady}
          opts={opts}
          sandbox="allow-same-origin"
        />

        {/* <Button onClick={this.play.bind(this)}>Play</Button>
            <Button>Pause</Button> */}
        <Button onClick={this.onPlayVideo}>Play</Button>
        <Button onClick={this.onPauseVideo}>Pause</Button>
        <Button onClick={this.seekSegment.bind(this)}>Segment Start</Button>

        <Button
          onClick={() => {
            markInAction(this.state.player.getCurrentTime());
            checkMarks();
          }}
        >
          In-Mark
        </Button>
        <Button
          onClick={() => {
            markOutAction(this.state.player.getCurrentTime());
            checkMarks();
          }}
        >
          Out-Mark
        </Button>
        <Button onClick={() => {}}>full-play</Button>
      </div>
    );
  }
}

export default connect(
  state => ({
    movies: state.playerReducer
  }),
  dispatch => ({
    actions: {
      checkMarks: () => dispatch(actions.checkMarks()),
      markInAction: milliseconds =>
        dispatch(actions.markInAction(milliseconds)),
      markOutAction: milliseconds =>
        dispatch(actions.markOutAction(milliseconds)),
      startPlaying: () => dispatch(actions.startPlaying())
    }
  })
)(Player);
