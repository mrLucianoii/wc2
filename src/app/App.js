import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { isLoadingAction as actions } from "../store/actions";
import Player from "../features/player/Player";
import logo from "./logo.svg";
import "./App.css";
import "./index.css";

class App extends Component {
  componentDidMount() {
    const { loaderAction } = this.props;
    loaderAction();
  }
  render() {
    const { loading } = this.props;
    return (
      <div className="App">
        {!loading && <h1>Loading...</h1>}
        {loading && (
          <Fragment>
            <header className="App-header">
              <img src={logo} className="App-logo" alt="logo" />
              WeCliques say's, "Hello World"
              <Player />
            </header>
          </Fragment>
        )}
      </div>
    );
  }
}

export default connect(
  state => ({
    loading: state.featureReducer.isLoading
  }),
  dispatch => ({
    loaderAction: () => dispatch(actions())
  })
)(App);
