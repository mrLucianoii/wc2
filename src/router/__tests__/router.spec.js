import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Router, { NoMatch } from '../index';
import App from '../../app/App';
import Player from '../../features/player/Player';
import renderer from "react-test-renderer";
import { Route } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() })

let pathMap = {};
describe('404 Page:', () => {
    it('Mounts', () => {

        const page404 = renderer.create(<NoMatch />).toJSON();
        expect(page404).toMatchSnapshot();
    })
});

describe('routes using array of routers', () => {
  beforeAll(() => {
    const component = shallow(<Router/>);
    pathMap = component.find(Route).reduce((pathMap, route) => {
        const routeProps = route.props();
        pathMap[routeProps.path] = routeProps.component;
        return pathMap;
      }, {});
  })
  it('should show Home component for / router (getting array of routes)', () => {

    expect(pathMap['/']).toBe(App);
  })
  it('should show YouTube player for /player router', () => {
    expect(pathMap['/player']).toBe(Player);
  })
  it('should show No match component for route not defined', ()=>{
      expect(pathMap['undefined']).toBe(NoMatch);
  })
})