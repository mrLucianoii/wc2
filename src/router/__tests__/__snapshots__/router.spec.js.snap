// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page: Mounts 1`] = `
<div>
  404 page
</div>
`;
